language/tools/vm-genesis/src/lib.rs

```diff
@@ -593,3 +597,20 @@ fn verify_genesis_write_set(
         "Discovery set in emitted event does not match discovery set fed into genesis transaction",
     );
 }
+
+// `StateView` has no data given we are creating the genesis
+struct EmptyStateView;
```

Comment:
Nit: maybe call this `GenesisStateView` instead of `EmptyStateView` since this should only be used in genesis?

Reply:
Yeah, `is_genesis` returns true so I agree with @tzakian